charter.md

```diff
@@ -53,15 +53,207 @@ The following responsibilities belong to the Steering Committe, but may be deleg
+processes set forth in the committee's [operations documentation][changes].
+
+[changes]: /operations/changes.md
+
+## Membership
+
+### Composition
+
+The Steering Committee is composed of seven (7) members.
+
+### Resignation
+
+If a committee member chooses not to continue in their role, for whatever
+self-elected reason, they must notify the committee in writing. As a courtesy,
+such notifications should be given at least 30 calendar days in advance of
+their departure.
+
+The committee can choose to accept a member's resignation and carry out
+offboarding procedures at any time following the written notice.
+
+### Removal
+
+<!--
+
+STEERING CONSIDERATIONS
+
+Removal
+
+- [ ] Special election for specific removal scenarios
+  - What is the perception of having a committee remove a representative that was elected by the community?
+- [ ] Codify the voting process and channels for removal scenarios
+- [ ] Consider CoCC guidance on removals as prior art: https://github.com/kubernetes/community/blob/master/committee-code-of-conduct/charter.md#removal
+- [ ] Consider prior art in changes to maintainership e.g., SPIFFE/SPIRE (https://github.com/spiffe/spire/blob/main/MAINTAINERS.md#changes-in-maintainership)
+
+COMMUNITY CONSIDERATIONS
+
+No Confidence
+
+- [ ] Provide guidelines on "no confidence"
+  - Process transparency could be considered inversely proportional to the importance of these guidelines
+
+Removal
+
+- [ ] Scope of removal of members (one, many)
+- [ ] Precedence for removal scenarios
+- [ ] What details of removal scenarios should be public/private?
+  - Call for removal?
+  - Reason for removal?
+  - Vote on removal?
+  - What are the public artifacts of the removal process?
+- [ ] What details of election results should be public/private?
+- [ ] Consequences of removal scenarios
+  - Are individuals ineligible for re-election?
+    - Should community members have an opinion in this decision?
+- [ ] When is removal appropriate?
+- [ ] How should CoC concerns be considered in the removal process? Should they be?
+- [ ] Guard against "popularity contests" or "burning at the person" in committee voting scenarios
+- [ ] Are there antitrust implications in the removal process?
+- [ ] Special elections may not have representative turnout
+  - Define guidance on acts and thresholds to consider in a vote for removal
+- [ ] Is removal an extraordinary power to grant?
+  - "Policies like this have to work for the good of the community both when used well and when used poorly."
+- [ ] Define removal process guidelines
+  - "brief rationale for removal" + short time + "brief rebuttal"
+  - Offer resignation as an option
+
+-->
+
+### No confidence
+
+In the event a Steering Committee member receives a vote of no confidence from
+two-thirds of the remaining Steering Committee members, that person must either
+resign or call a special election in which the project community will vote on
```

Comment:
I appreciate that this swings pretty hard in the "more transparent" direction, but (and this is just my opinion, which I won't fight too hard for) I think it goes too far and drags things out in a way that isn't useful.

"must resign" is an oxymoron.  Presumably a straw-poll will be taken amongst steering.  If the person in question knows they will lose that vote, and still chooses not to resign, we don't have a removal process, we have a recall process.  This will require significantly more time, energy, and drama - likely including campaigning which will do more damage to all parties INCLUDING the project writ large.

If that is the path you all want to take, I will stop debating it.  My goal is not to make it all more difficult.

Reply:
@justaugustus @parispittman i think we need to talk about this. There was a simpler path in the discussion above https://github.com/kubernetes/steering/pull/248#discussion_r911469884 that we should noodle on.

Reply:
I'm inclined to agree with @thockin and @dims. Wrangling the normal SC election is already a time-intensive process. Doing it ad-hoc would REALLY draw things out. I think its worth discussing a more streamlined path.

Reply:
Sounds good! This was an attempt to incorporate some of the feedback on what was initially written and thoughts about special elections (like https://github.com/kubernetes/steering/pull/248#issuecomment-1171036764) and then blend the two approaches.

I'll noodle on simpler paths (like https://github.com/kubernetes/steering/pull/248#discussion_r911469884) while more comments roll in.

Reply:
+1 - I don't think the special elections route is useful and drags out a process that realistically needs to be concluded as soon as possible.

Reply:
For the record, when I argue for "transparency", I am not calling for a trial or even evidence, really.  Simply a statement of rationale and maybe a rebuttal.

My mental model is something like:

* person X proposes a motion for a vote of no confidence, with a 100 words or less statement (public)
* person Y seconds
* the motion is carried
* the subject of the motion, if they so desire, has N days to make a 100 words or less rebuttal (public)
  - N is not less than 7 and not more than 28, as determined by the committee to accomodate travel or other plans person Z may have
* after N days a public vote of the committee is held

No special election, no demanding evidence.  It's not a trial.  I'm 50/50 on public comment, but if so, it is non-binding and voluntary (not sure if anonymous or not?).

If the community overall thinks there was injustice done and that the committee does not represent their values, there are regular elections.

Reply:
The only thing I'd add to that is that the timeline should give the subject time to produce a rebuttal well before N, it would not go well to produce a rebuttal the day before the vote. But that's (hopefully obvious to the subject) just strategy.